### PR TITLE
Invalid Graph on Combobox

### LINKED ISSUE #433: Invalid Graph on Combobox
When adding a new usual graph from the template on a filtered .g6 file, the software adds to the combobox 2 items:
- an empty graph
- the actual graph the person created

When clicking on the blank graph, the software throws an error and adds the label "Invalid Graph" to it.

Investigating, it was discovered that the filtered .g6 file has a final line break, and when adding another graph, the blank line is kept, thus creating the invalid graph.

> **Note**
> This problem is related to the issue solved on #397

### Example
```g6
GQhTVS
GQjnV[
GQjn^[
GQzn^[
GTm|~w
GTm|~{

Esa?
``` 

### Demonstration
Graph-Filter-wsgt_graphs g6-2023-10-29-21-04-41

### PR DESCRIPTION
**Pull request by rosaigor138**
Closes #433

### CODE REVIEW

**Comment by atilaajones** [comment]
You should include the issue number in the description. This way we'll do a review to understand the motivation behind the update. Also try to be more specific in the commit name, not just "fixed"

**Review by atilaajones — CHANGES_REQUESTED** [review]
This change solved the problem. But I think you should also change the source of the problem: filtering should not return a file with a blank last line. The same goes for inserting the usual new graph, which also inserts a blank line.

**Comment by LBeghini** [comment]
> This change solved the problem. But I think you should also change the source of the problem: filtering should not return a file with a blank last line. The same goes for inserting the usual new graph, which also inserts a blank line.

@atilaajones the blank line after filtering was an strategy proposed to fix the issue #397 . We can see the changes on the PR #403

**Comment by fsoupimenta** [comment]
I have two observations about this PR:

1. On `Linux` (more precisely on Ubuntu 22.04.3 LTS) when a file with a blank line at the end is opened, the software already reads this line and throws it into the combobox, leaving the graph invalid. It's worth noting that this behavior only occurs on `Linux`.
2. When you click on an invalid graph in the combobox, the graph counter decreases, leaving duplicates. This behavior probably comes from solving #408 (we can see the changes on #412). An issue will be opened to resolve this bug

**Comment by atilaajones** [comment]
Test the reported behavior, then try to solve it in PR #444

**Comment by rosaigor138** [comment]
> Test the reported behavior, then try to solve it in PR #444 @rosaigor138

I don't understand why that I need to make that changes on the change graph file PR

File: source/controller/project_controller.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def on_check_condition(self):
 
     def on_invalid_graph_display_alert(self):
         index = self.project_tool_bar.combo_graphs.currentIndex()
-        self.project_tool_bar.combo_graphs.setItemText(index, f'Graph {index} - Invalid')
+        self.project_tool_bar.combo_graphs.setItemText(index, f'Graph {index+1} - Invalid')
```
Comment (rosaigor138) [inline source/controller/project_controller.py:302]: @fsoupimenta  I found the problem, when the invalid graph window was called, the combobox counter was reogarnized, but the index received was from 0 to n-1 unlike how our combobox works.

**Review by LBeghini — APPROVED** [review]
Working on macOS Sonoma version 14.1.1 Chip Apple M1

**Review by fsoupimenta — CHANGES_REQUESTED** [review]
@rosaigor138 The blank line at the end is still read by the software in ubuntu

**Review by atilaajones — APPROVED** [review]
@fsoupimenta. I think this PR can be merged, right?

**Pull request closed without merging**